src/render/Renderer.cpp

```diff
@@ -484,17 +484,25 @@ void CHyprRenderer::renderWorkspaceWindows(PHLMONITOR pMonitor, PHLWORKSPACE pWo
 
     EMIT_HOOK_EVENT("render", RENDER_PRE_WINDOWS);
 
-    // Non-floating main
+    std::vector<std::reference_wrapper<PHLWINDOW const>> windows;
```

Comment:
instead of a ref wrapper, why not just use a WP? Would it be that much slower?

Reasoning is, since we never render a window twice, you could set the weak_ptr to a nullptr after every render and that would provide a very fast bail-out in further loops (if (!w) continue)

Reply:
oh true thats also one way to tackle it. nah WP is fine, i just tried different things and got down to the least expensive way, but didnt consider that way. heh.

Reply:
yes. In the first loop if (!w) is always false tho can be dropped